pandas/core/window/rolling.py

```diff
@@ -875,6 +875,14 @@ def _gotitem(self, key, ndim, subset=None):
             subset = self.obj.set_index(self._on)
         return super()._gotitem(key, ndim, subset=subset)
 
+    def aggregate(self, func, *args, **kwargs):
+        result = super().aggregate(func, *args, **kwargs)
+        if not self._as_index:
```

Comment:
I think this should live in `BaseWindowGroupby` so added this aggregate function. After adding, mypy complained about incompatibility of base classes for classes inheriting `BaseWindowGroupby`. Adding aggregate function to those solves the issue happy to extend their docs if required or implement another workaround if preferred over this